terraform/sqs/bind/data.tf

```diff
@@ -1,15 +1,7 @@
-data "aws_iam_policy_document" "user_policy" {
-  dynamic "statement" {
-    for_each = local.queue_policy
-    content {
-      sid       = try(statement.value.sid, null)
-      actions   = try(statement.value.actions, null)
-      resources = try(statement.value.resources, null)
-    }
-  }
-}
-
 locals {
+  key_ids_list = compact(split(",", var.kms_all_key_ids))
+  has_key_ids  = length(local.key_ids_list) != 0
```

Comment:
nitp: Declering these closer to where they are used makes it easier to read navigate this file.

Reply:
Sounds good to me.